### PR TITLE
Stabilize suggestion applicability field in json output

### PR DESCRIPTION
**Pull request by Manishearth**
cc @killercup

r? @nrc

### CODE REVIEW

**Comment by Manishearth** [comment]
We should start work on stabilizing the field now so that it's ready for the edition. This also makes it easier for the rustfix preview.


It's an extra field in JSON, I don't think this is something we need to worry much about stabilizing.

File: src/libsyntax/json.rs [hunk 1]
```diff
@@ -137,7 +131,6 @@ struct DiagnosticSpan {
     /// that should be sliced in atop this span.
     suggested_replacement: Option<String>,
     /// If the suggestion is approximate
-    #[rustc_serialize_exclude_null]
     suggestion_applicability: Option<Applicability>,
```
Comment (killercup) [inline src/libsyntax/json.rs]: Am I reading the code below correctly as: `suggestion_applicability` is always `Some` when `suggested_replacement` is `Some`? If so, we might want to document this here.

**Review by killercup — APPROVED** [review]
🎉

**Comment by Manishearth** [comment]
@bors r-

I'll need to update the ui tests

**Comment by pietroalbini** [comment]
Ping from triage @Manishearth! It's been a while since we heard from you, will you have time to work on this again soon?

**Pull request merged**